frontends/concrete-python/examples/prime-match/prime-match-semi-honest.py

```diff
@@ -0,0 +1,120 @@
+import random
+import time
+
+import numpy as np
+
+from concrete import fhe
+
+# Users can change these settings
+NUMBER_OF_SYMBOLS = 10
+MINIMUM_ORDER_QUANTITY = 1
+MAXIMUM_ORDER_QUANTITY = 50
+
+assert 0 < MINIMUM_ORDER_QUANTITY < MAXIMUM_ORDER_QUANTITY
+
+
+def prime_match(
+    bank_order_quantities,
+    client_order_quantities,
+):
+    with fhe.tag("calculating-matching-order-quantity"):
+        return np.minimum(bank_order_quantities, client_order_quantities)
+
+
+inputset = [
+    (
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY, MAXIMUM_ORDER_QUANTITY, size=(NUMBER_OF_SYMBOLS * 2,)
+        ),
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY, MAXIMUM_ORDER_QUANTITY, size=(NUMBER_OF_SYMBOLS * 2,)
+        ),
+    )
+    for _ in range(1000)
+]
+configuration = fhe.Configuration(
+    enable_unsafe_features=True,
+    use_insecure_key_cache=True,
+    insecure_key_cache_location=".keys",
+    fhe_simulation=True,
+    min_max_strategy_preference=fhe.MinMaxStrategy.ONE_TLU_PROMOTED,
+    show_progress=True,
+    progress_tag=True,
+)
+
+
+# Only the quantities are encrypted, on both client and bank sides
+compiler = fhe.Compiler(
+    prime_match,
+    {
+        "bank_order_quantities": "encrypted",
+        "client_order_quantities": "encrypted",
+    },
+)
+circuit = compiler.compile(inputset, configuration)
+
+
+print()
```

Comment:
Maybe create a utility function to run and log steps
```
def run_and_log(message, f):
    start = time.time()
    f()    
    end = time.time()
    print(f"{message} took: {end - start:.3f} seconds")
```

Reply:
I could do that but it's done only one per .py. Isn't it over engineering here?